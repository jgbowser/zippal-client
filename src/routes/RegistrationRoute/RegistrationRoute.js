import React, {Component} from 'react'
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'
import Loading from '../../components/Loading/Loading'

class RegistrationRoute extends Component {
  constructor(props) {
    super(props)
    this.state = {
      loading: false
    }
  }
  static defaultProps = {
    history: {
      push:() => {},
    },
  }

  handleRegistrationSuccess = () => {
    const {history} = this.props
    history.push('/profile')
  }

  toggleLoading = () => {
    this.setState({
      loading: !this.state.loading
    })
  }

  render(){
    return (
      <section className='registration_section'>
        <h2>Sign up</h2>
        { this.state.loading === false ? <RegistrationForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
          toggleLoading={this.toggleLoading}
        /> : <Loading />}
      </section>
    );
  }
}

export default RegistrationRoute