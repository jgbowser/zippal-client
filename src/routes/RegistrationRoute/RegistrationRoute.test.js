import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import RegistrationRoute from './RegistrationRoute'
import renderer from 'react-test-renderer'

describe('RegistrationRoute Component', () => {
  it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<BrowserRouter><RegistrationRoute /></BrowserRouter>, div)
  ReactDOM.unmountComponentAtNode(div)
  })
  it(`takes a snapshot of the RegistrationRoute`,()=>{
    const tree = renderer
    .create(<BrowserRouter><RegistrationRoute /></BrowserRouter>)
    expect(tree).toMatchSnapshot()
  })
})