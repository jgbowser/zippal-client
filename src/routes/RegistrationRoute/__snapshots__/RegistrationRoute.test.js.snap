// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationRoute Component takes a snapshot of the RegistrationRoute 1`] = `
<section
  className="registration_section"
>
  <h2>
    Sign up
  </h2>
  <form
    className="user_form"
    onSubmit={[Function]}
  >
    <legend>
      <div
        role="alert"
      />
      <p>
        <span
          className="required"
        >
          *
        </span>
         indicates required field
      </p>
      <div>
        <label
          htmlFor="registration-name-input"
        >
          Display Name
          <span
            className="required"
          >
            *
          </span>
        </label>
        <input
          id="registraion-name-input"
          name="name"
          required={true}
        />
      </div>
      <div>
        <label
          htmlFor="registration-username-input"
        >
          Username
          <span
            className="required"
          >
            *
          </span>
        </label>
        <input
          id="registration-username-input"
          name="username"
          required={true}
        />
      </div>
      <div>
        <label
          htmlFor="registration-password-input"
        >
          Password
          <span
            className="required"
          >
            *
          </span>
        </label>
        <input
          id="registration-password-input"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <footer>
        <button
          type="submit"
        >
          Sign up
        </button>
         
        <a
          href="/login"
          onClick={[Function]}
        >
          Already have an account?
        </a>
      </footer>
    </legend>
  </form>
</section>
`;
